build.cake

```diff
@@ -0,0 +1,581 @@
+
+/// <summary>
+///  Class representing build.json
+/// </summary>
+public class BuildPlan
+{
+    public IDictionary<string, string[]> TestProjects { get; set; }
+    public string BuildToolsFolder { get; set; }
+    public string ArtifactsFolder { get; set; }
+    public string DotNetFolder { get; set; }
+    public string DotNetInstallScriptURL { get; set; }
+    public string DotNetChannel { get; set; }
+    public string DotNetVersion { get; set; }
+    public string[] Frameworks { get; set; }
+    public string[] Rids { get; set; }
+    public string MainProject { get; set; }
+}
+
+var buildPlan = DeserializeJsonFromFile<BuildPlan>($"{workingDirectory}/build.json");
+
+// Folders and tools
+var dotnetFolder = $"{workingDirectory}/{buildPlan.DotNetFolder}";
+var dotnetcli = $"{dotnetFolder}/cli/dotnet";
+var toolsFolder = $"{workingDirectory}/{buildPlan.BuildToolsFolder}";
+var xunitRunner = "xunit.runner.console";
+
+var sourceFolder = $"{workingDirectory}/src";
+var testFolder = $"{workingDirectory}/tests";
+
+var artifactFolder = $"{workingDirectory}/{buildPlan.ArtifactsFolder}";
+var publishFolder = $"{artifactFolder}/publish";
+var logFolder = $"{artifactFolder}/logs";
+var packageFolder = $"{artifactFolder}/package";
+
+/// <summary>
+///  Retrieve the default local RID from .NET CLI.
+/// </summary>
+/// <param name="dotnetcli">Full path to .NET CLI binary</param>
+/// <returns>Default RID</returns>
+string GetLocalRuntimeID(string dotnetcli)
+{
+    var process = StartAndReturnProcess(dotnetcli, 
+        new ProcessSettings
+        { 
+            Arguments = "--version",
+            RedirectStandardOutput = true
+        });
+    process.WaitForExit();
+    foreach (var line in process.GetStandardOutput())
+    {
+        if (!line.Contains("Runtime Id"))
+        {
+            continue;
+        }
+        var colonIndex = line.IndexOf(':');
+        return line.Substring(colonIndex + 1).Trim();
+    }
+    throw new Exception("Failed to get default RID for system");
+}
+
+/// <summary>
+///  Match the local RID with the ones specified in build.json.
+///  Return exact match if found.
+///  Return first OS match (without version number) otherwise.
+///  Report error if no OS match found.
+/// </summary>
+/// <param name="dotnetcli">Full path to .NET CLI binary</param>
+/// <param name="buildPlan">BuildPlan from build.json</param>
+/// <returns>Matched RID</returns>
+string MatchLocalRuntimeID(string dotnetcli, BuildPlan buildPlan)
+{
+    var localRuntime = GetLocalRuntimeID(dotnetcli);
+    if (buildPlan.Rids.Contains(localRuntime))
+    {
+        return localRuntime;
+    }
+    else
+    {
+        return buildPlan.Rids.First(runtime => {
+            var localRuntimeWithoutVersion = Regex.Replace(localRuntime, "(\\d|\\.)*-", "-");
+            var runtimeWithoutVersion = Regex.Replace(runtime, "(\\d|\\.)*-", "-");
+            return localRuntimeWithoutVersion.Equals(runtimeWithoutVersion);
```

Comment:
Then it won't build both x86 and x64 versions on windows.  I guess I could use that to further limit it for linux and mac, at least until (if?) x86 versions become a thing.